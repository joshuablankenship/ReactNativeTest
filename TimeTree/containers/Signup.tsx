import React, {Component} from 'react';
import {Image, Text, TextInput, TouchableOpacity, View} from 'react-native';
import RBSheet from 'react-native-raw-bottom-sheet';
import {styles} from '../styles/SignupStyles';
import CreateProfile from './CreateProfile';
interface Props {
  navigation: any;
}
export default class Signup extends Component<Props> {
  createProfileSheet: RBSheet | null | undefined;
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      isPopupTrue: true,
      errorEmail: '',
      errorPassword: '',
      hasErrors: false,
    };
  }
  validate = text => {
    console.log(text);
    let reg = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
    if (reg.test(text) === false) {
      return false;
    } else {
      return true;
    }
  };
  _checkForErrors = () => {
    let hasErrorsEmail = this.validate(this.state.email) == false || this.state.email.length ==0 
    let hasErrorsPassword = this.state.password.length < 8|| this.state.password.length > 32;
    let emailError = '';
    let passwordError= '';
    if (hasErrorsEmail && hasErrorsPassword) {
      emailError = 'Please Enter a valid email address';
      passwordError= 'Please Enter a valid password';
    }else if(hasErrorsPassword && !hasErrorsEmail){
      emailError = '';
      passwordError= 'Please Enter a valid password';
    }else if(hasErrorsEmail && !hasErrorsPassword){
      emailError = 'Please Enter a valid email address';
      passwordError= '';
    }
    console.log(this.state.password + this.state.errorEmail)

    this.setState({errorEmail: emailError, errorPassword: passwordError});

    if(!hasErrorsEmail && !hasErrorsPassword) {
      this.createProfileSheet.open();

    }
  };
  _onPressButtonGoogle = async () => {
    this.props.navigation.navigate('Main');
  };
  _openSignUpSheet = () => {
    this._checkForErrors();
  };
  _closeSignUpSheet = () => {
    this.props.closeSignupSheet();
  };
  _openProfileSheet = () => {
    this._checkForErrors();
  };
  _closeProfileSheet = () => {
    this.createProfileSheet.close();
  };
  render() {
    return (
      <View>
        <View style={styles.sheetHeader}>
          <TouchableOpacity
            style={styles.text}
            onPress={this._closeSignUpSheet}>
            <Image
              style={styles.a_logo}
              source={require('../assets/images/x.png')}
            />
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.text}
            onPress={this._closeSignUpSheet}>
            <Text style={styles.text}> Later</Text>
          </TouchableOpacity>
        </View>
        <Text style={styles.title}> Sign up</Text>
        <View style={styles.buttonContainer}>
          <TextInput
            style={
              this.state.errorEmail.length > 0
                ? styles.textinputError
                : styles.textinput
            }
            placeholder="Email Address"
            autoCorrect={false}
            onChangeText={text => this.setState({email: text})}
            value={this.state.email}
          />
          <Text style={styles.error}>{this.state.errorEmail}</Text>
          <TextInput
            style={
              this.state.errorPassword.length > 0
                ? styles.textinputError
                : styles.textinput
            }
            secureTextEntry
            placeholder="8 - 32 character password"
            onChangeText={text => this.setState({password: text})}
            value={this.state.password}
          />
           <Text style={styles.error}>{this.state.errorPassword}</Text>
          
          <TouchableOpacity
            style={styles.buttonGreen}
            onPress={this._openProfileSheet}>
            <Text style={styles.buttonText}>Create Account</Text>
          </TouchableOpacity>
          <Text style={styles.textCenter}>or</Text>
          <TouchableOpacity style={styles.buttonFacebook}>
            <Image
              style={styles.f_logo}
              source={require('../assets/images/facebook_logo.png')}
            />
            <Text style={styles.buttonText}> Login with Facebook</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.buttonApple}
            // onPress={() => this.RBSheet.open()}
          >
            <Image
              style={styles.a_logo}
              source={require('../assets/images/apple_logo.png')}
            />
            <Text style={styles.buttonText}> Sign in with Apple</Text>
          </TouchableOpacity>
        </View>
        <View>
          <RBSheet
            ref={ref => {
              this.createProfileSheet = ref;
            }}
            height={800}
            openDuration={250}
            customStyles={{
              container: {
                borderTopLeftRadius: 10,
                borderTopRightRadius: 10,
              },
            }}>
            <CreateProfile
              navigation={this.props.navigation}
              openProfileSheet={this._openProfileSheet}
              closeProfileSheet={this._closeProfileSheet}
              closeSignupSheet={this._closeSignUpSheet}
              closeLoginSheet={this.props.closeLoginSheet}
            />
          </RBSheet>
        </View>
      </View>
    );
  }
}
