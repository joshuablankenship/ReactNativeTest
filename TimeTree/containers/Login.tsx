import React, {Component} from 'react';
import SplashScreen from 'react-native-splash-screen';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import RBSheet from 'react-native-raw-bottom-sheet';
import {styles} from '../styles/LoginStyles';
import Bottom from './Bottom';
import Signup from './Signup';
interface Props {
  navigation: any;
}
interface State {
  isPopupTrue: boolean;
}
export default class Login extends Component<Props, State> {
  constructor(props: Props | Readonly<Props>) {
    super(props);
    this.state = {
      isPopupTrue: false,
    };
  }
  RBSheet: RBSheet | null | undefined | any;
  signupSheet: RBSheet | null | undefined | any;
  componentDidMount() {
    SplashScreen.hide();
  }
  _openSignUpSheet = () => {
    this.signupSheet.open();
  };
  _closeSignUpSheet = () => {
    this.signupSheet.close();
  };
  _closeLoginSheet = () => {
    this.RBSheet.close();
  };
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.alternativeLayoutButtonContainer}>
          <Image
            style={styles.image}
            source={require('../assets/images/logo.png')}
          />
        </View>
        <View style={styles.buttonContainer}>
          <TouchableOpacity
            style={styles.buttonGreen}
            onPress={() => this.RBSheet.open()}>
            <Text style={styles.buttonText}>Login</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.button}
            onPress={this._openSignUpSheet}>
            <Text style={styles.buttonTextGreen}>Get Started</Text>
          </TouchableOpacity>
        </View>
        <View style={styles.textContainer}>
          <Text style={styles.text}>
            By selecting one or the other you are agreeing to our{' '}
            <Text style={styles.link}>Terms of Service</Text> {'& '}
            <Text style={styles.link}>Privacy Policy </Text>.
          </Text>
        </View>
        <View style={styles.container}>
          <RBSheet
            ref={ref => {
              this.RBSheet = ref;
            }}
            height={300}
            openDuration={250}
            closeOnDragDown
            customStyles={{
              container: {
                borderTopLeftRadius: 10,
                borderTopRightRadius: 10,
              },
            }}>
            <Bottom
              navigation={this.props.navigation}
              closeLoginSheet={this._closeLoginSheet}
            />
          </RBSheet>
          <View />
          <View style={styles.container}>
            <RBSheet
              ref={ref => {
                this.signupSheet = ref;
              }}
              height={800}
              openDuration={250}
              customStyles={{
                container: {
                  borderTopLeftRadius: 10,
                  borderTopRightRadius: 10,
                },
              }}>
              <Signup
                navigation={this.props.navigation}
                openSignupSheet={this._openSignUpSheet}
                closeSignupSheet={this._closeSignUpSheet}
                closeLoginSheet={this._closeLoginSheet}
              />
            </RBSheet>
          </View>
        </View>
      </View>
    );
  }
}
