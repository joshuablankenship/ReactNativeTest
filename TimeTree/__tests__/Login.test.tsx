import {shallow, ShallowWrapper} from 'enzyme';
import React from 'react';
import {Image, TouchableOpacity} from 'react-native';
import Login from '../containers/Login';
import Signup from '../containers/Signup';
const createTestProps = (props: any) => ({
  navigation: {
    navigate: jest.fn(),
  },
  closeLoginSheet: jest.fn(),
  ...props,
});
jest.mock('react-native-splash-screen', () => {
  return {
    hide: jest.fn(),
    show: jest.fn(),
  };
});
describe('Test Login Screen', () => {
  let wrapper: ShallowWrapper;
  let props: any;
  wrapper = shallow(<Login {...props} />);
  props = createTestProps({
    navigate: () => {},
    closeLoginSheet: () => {},
  });
  it('It should match initial snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('It should display three touchable opacity components', () => {
    expect(wrapper.find(TouchableOpacity).length).toEqual(2);
  });
  it('It should display one logo image', () => {
    expect(wrapper.find(Image).length).toEqual(1);
  });
  it('It should contain one sign up view', () => {
    expect(wrapper.find(Signup).length).toEqual(1);
  });
});
