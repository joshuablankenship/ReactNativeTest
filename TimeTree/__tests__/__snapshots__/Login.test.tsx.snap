// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login Screen It should match initial snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingTop": 200,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "margin": 20,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/logo.png",
        }
      }
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "height": 250,
          "justifyContent": "center",
          "marginBottom": 50,
          "width": 250,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "marginTop": 50,
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2FCC87",
          "borderRadius": 5,
          "color": "white",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginBottom": 10,
          "marginLeft": 40,
          "marginRight": 40,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        Login
      </Text>
    </ForwardRef>
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#2FCC87",
          "borderRadius": 5,
          "borderWidth": 0.25,
          "color": "#2FCC87",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "marginBottom": 10,
          "marginLeft": 40,
          "marginRight": 40,
          "padding": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#2FCC87",
            "fontWeight": "bold",
          }
        }
      >
        Get Started
      </Text>
    </ForwardRef>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginLeft": 40,
        "marginRight": 40,
      }
    }
  >
    <Text
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      By selecting one or the other you are agreeing to our
       
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "#2FCC87",
            "flex": -1,
            "flexDirection": "row",
            "fontWeight": "bold",
            "justifyContent": "center",
            "marginTop": 5,
            "textDecorationLine": "underline",
          }
        }
      >
        Terms of Service
      </Text>
       
      & 
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "#2FCC87",
            "flex": -1,
            "flexDirection": "row",
            "fontWeight": "bold",
            "justifyContent": "center",
            "marginTop": 5,
            "textDecorationLine": "underline",
          }
        }
      >
        Privacy Policy 
      </Text>
      .
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingTop": 200,
      }
    }
  >
    <RBSheet
      animationType="none"
      closeDuration={200}
      closeOnDragDown={true}
      closeOnPressBack={true}
      closeOnPressMask={true}
      customStyles={
        Object {
          "container": Object {
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
          },
        }
      }
      dragFromTopOnly={false}
      height={300}
      keyboardAvoidingViewEnabled={true}
      minClosingHeight={0}
      onClose={null}
      onOpen={null}
      openDuration={250}
    >
      <Bottom
        closeLoginSheet={[Function]}
      />
    </RBSheet>
    <View />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingTop": 200,
        }
      }
    >
      <RBSheet
        animationType="none"
        closeDuration={200}
        closeOnDragDown={false}
        closeOnPressBack={true}
        closeOnPressMask={true}
        customStyles={
          Object {
            "container": Object {
              "borderTopLeftRadius": 10,
              "borderTopRightRadius": 10,
            },
          }
        }
        dragFromTopOnly={false}
        height={800}
        keyboardAvoidingViewEnabled={true}
        minClosingHeight={0}
        onClose={null}
        onOpen={null}
        openDuration={250}
      >
        <Signup
          closeLoginSheet={[Function]}
          closeSignupSheet={[Function]}
          openSignupSheet={[Function]}
        />
      </RBSheet>
    </View>
  </View>
</View>
`;
